sphinx/ext/viewcode.py

```diff
@@ -141,6 +141,23 @@ def missing_reference(app, env, node, contnode):
     return None
 
 
+def split_code_lines(lines):
```

Comment:
I feel this code is a bit tricky. To be our code robust, we need to make a custom Formatter for pygments. I'll take a look in this weekend.